src/algebra/pointwise.lean

```diff
@@ -163,6 +163,55 @@ begin
       simp * } }
 end
 
+@[to_additive set.pointwise_add_eq_Union_add_left]
+lemma pointwise_mul_eq_Union_mul_left [has_mul α] {s t : set α} : s * t = ⋃a∈s, (λx, a * x) '' t :=
+by { ext y; split; simp only [mem_Union]; rintros ⟨a, ha, x, hx, ax⟩; exact ⟨a, ha, x, hx, ax.symm⟩ }
+
+@[to_additive set.pointwise_add_eq_Union_add_right]
+lemma pointwise_mul_eq_Union_mul_right [has_mul α] {s t : set α} : s * t = ⋃a∈t, (λx, x * a) '' s :=
+by { ext y; split; simp only [mem_Union]; rintros ⟨a, ha, x, hx, ax⟩; exact ⟨x, hx, a, ha, ax.symm⟩ }
+
+@[to_additive set.pointwise_add_ne_empty]
+lemma pointwise_mul_ne_empty [has_mul α] {s t : set α} : s ≠ ∅ → t ≠ ∅ → s * t ≠ ∅ :=
+begin
+  simp only [ne_empty_iff_exists_mem],
+  rintros ⟨x, hx⟩ ⟨y, hy⟩,
+  exact ⟨x * y, mul_mem_pointwise_mul hx hy⟩
+end
+
+@[simp, to_additive univ_add_univ]
+lemma univ_pointwise_mul_univ [monoid α] : (univ : set α) * univ = univ :=
+begin
+  have : ∀x, ∃a b : α, x = a * b := λx, ⟨x, ⟨1, (mul_one x).symm⟩⟩,
+  show {a | ∃ x ∈ univ, ∃ y ∈ univ, a = x * y} = univ,
+  simpa [eq_univ_iff_forall]
+end
+
+section is_mul_hom
+open is_mul_hom
+
+variables [has_mul α] [has_mul β] (m : α → β) [is_mul_hom m]
+
+@[to_additive is_add_hom.image_add]
+lemma image_pointwise_mul (s t : set α) : m '' (s * t) = m '' s * m '' t :=
```

Comment:
There is also https://github.com/leanprover-community/mathlib/blob/e229bf465f96f84406bcc324513a1b9ef5d4e7fb/src/algebra/pointwise.lean#L247 in this file:
```lean
variables [monoid α] [monoid β] [is_monoid_hom f]

def pointwise_mul_image_is_semiring_hom : is_semiring_hom (image f) :=
```
Is that sufficient for your purpose? Or otherwise, could you refactor it to use this lemma?